client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,95 @@
+import { Button } from '@wordpress/components';
+import { chevronLeft, chevronRight, Icon } from '@wordpress/icons';
+import { PropsWithChildren, ReactElement, useEffect, useRef, useState } from 'react';
+import './style.scss';
+import { Swiper as SwiperType } from 'swiper/types';
+
+interface ThemeCollectionProps {
+	collectionSlug: string;
+	title: string;
+	description: ReactElement;
+}
+
+export default function ThemeCollection( {
+	collectionSlug,
+	title,
+	description,
+	children,
+}: PropsWithChildren< ThemeCollectionProps > ): ReactElement {
+	const swiperInstance = useRef< SwiperType | null >( null );
+	const swiperContainerId = `swiper-container-${ collectionSlug }`;
+
+	const [ isSwiperLoaded, setSwiperLoaded ] = useState( false );
+	useEffect( () => {
+		if ( ! isSwiperLoaded || ! swiperInstance?.current || swiperInstance.current?.destroyed ) {
+			const el = document.querySelector( `#${ swiperContainerId }` ) as HTMLElement;
+			if ( el ) {
+				/**
+				 * We have to import the swiper package dynamically because it doesn't offer a CommonJS version. Because of this, the SSR build will fail.
+				 *
+				 * From a performance standpoint, it should not cause any issues; we are already loading the ThemeCards dynamically anyway.
+				 */
+				Promise.all( [
+					import( 'swiper' ),
+					import( 'swiper/modules' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/navigation' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/keyboard' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/mousewheel' ),
+				] ).then( ( values ) => {
+					const Swiper = values[ 0 ].default;
+					const { Navigation, Keyboard, Mousewheel } = values[ 1 ];
+
+					swiperInstance.current = new Swiper( el, {
+						cssMode: true,
+						mousewheel: true,
+						keyboard: {
+							enabled: true,
+							onlyInViewport: false,
+						},
+						navigation: {
+							nextEl: '.theme-collection__carousel-nav-button--next',
+							prevEl: '.theme-collection__carousel-nav-button--previous',
+						},
+						threshold: 5,
+						slideToClickedSlide: true,
+						slidesPerView: 'auto',
+						spaceBetween: 20,
+						modules: [ Navigation, Keyboard, Mousewheel ],
+					} );
+					setSwiperLoaded( true );
+				} );
+			}
+		}
+
+		return () => {
+			swiperInstance.current?.destroy();
+		};
+	}, [ swiperContainerId, isSwiperLoaded ] );
+
+	return (
+		<div className="theme-collection__container ">
+			<h2>{ title }</h2>
+			{ description }
```

Comment:
I'd recommend adding some specificity. I'm sure it will turn out to be useful. 🙂
```suggestion
			<h2 className="theme-collection__title">{ title }</h2>
			<div className="theme-collection__description">{ description }<div>
```

Reply:
100% will, I already did the same in https://github.com/Automattic/wp-calypso/pull/82519/files